### DISCUSSION TITLE
Shape Matching constraint: not using Q^-1 ??

### DISCUSSION THREAD

**Discussion by metanetsoftware** [InteractiveComputerGraphics/PositionBasedDynamics#161] [Q&A]
hi,
In the Shape Matching papers, they calculate A = PQ^-1.

I noticed that in https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/master/PositionBasedDynamics/PositionBasedDynamics.cpp , you build mat = P, and then on line 539 I saw: //mat = mat * invRestMat;

This means that you're using A = P (ie the Q^-1 term is removed entirely); I was wondering what the motivation for this was, since I can't find anything in any of the papers which mentions this. 

However: I've just implemented a 2D version of your approach, and it appears to work! So I'm really curious why Q^-1 isn't needed (and/or what it represents), and I was hoping you could shed some light on this.

Cheers,
Raigan

**Comment by janbender** [comment]
If you take a look in the original paper of Müller et al. "Meshless Deformations Based on Shape Matching" in Eq. 7. The matrix Q corresponds to the matrix A_qq in the paper. As mentioned in the paper "The second term Aqq is a symmetric matrix and, thus, contains only
scaling but no rotation." So there is no relevant information in this matrix and we just extract the rotation from the matrix A_pq.

**Reply by metanetsoftware** [reply]
ah, that makes sense -- thanks so much!
